Lib/test/test_datetime.py

```diff
@@ -8,10 +8,12 @@
 TESTS = 'test.datetimetester'
 
 try:
-    pure_tests = import_fresh_module(TESTS, fresh=['datetime', '_strptime'],
+    pure_tests = import_fresh_module(TESTS,
+                                     fresh=['datetime', '_pydatetime', '_strptime'],
                                      blocked=['_datetime'])
-    fast_tests = import_fresh_module(TESTS, fresh=['datetime',
-                                                   '_datetime', '_strptime'])
+    fast_tests = import_fresh_module(TESTS,
+                                     fresh=['datetime', '_datetime', '_strptime'],
+                                     blocked=['_pydatetime'])
```

Comment:
Can't leave comments for lines not directly affected by the PR, but I believe we now also need to add `_pydatetime` to the list below:

```python
for modname in ['datetime', '_datetime', '_pydatetime', '_strptime']:
    sys.modules.pop(modname, None)
```

(At least until we understand why that was necessary in the first place...